src/sentry/integrations/messaging.py

```diff
@@ -216,3 +237,174 @@ def build_handler(
         self, action: AlertRuleTriggerAction, incident: Incident, project: Project
     ) -> ActionHandler:
         return MessagingActionHandler(action, incident, project, self.spec)
+
+
+@control_silo_view
+class LinkingView(BaseView, ABC):
+    @property
+    @abstractmethod
+    def parent_messaging_spec(self) -> MessagingIntegrationSpec:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def provider(self) -> ExternalProviders:
+        raise NotImplementedError
+
+    @property
+    def provider_slug(self) -> str:
+        return self.parent_messaging_spec.provider_slug
+
+    @property
+    @abstractmethod
+    def salt(self) -> str:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def user_parameter(self) -> str:
+        raise NotImplementedError
+
+    # TODO: Replace thw two template properties below with base templates for all
+    #       integrations to use. Add service-specific parts to the context as needed.
+
+    @property
+    @abstractmethod
+    def confirmation_template(self) -> str:
+        """Path to the HTML template to render for a non-POST request."""
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def expired_link_template(self) -> str:
+        """Path to the HTML template to show when a link is expired."""
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def success_template(self) -> str:
+        """Path to the HTML template to show when an identity has been linked."""
+        raise NotImplementedError
+
+    @property
+    def success_metric(self) -> str | None:
+        """Optional analytics key to record on success."""
+        return None
```

Comment:
should we use something like this for success / failure of linking / unlinking?

Reply:
I generalized this in https://github.com/getsentry/sentry/pull/75899 (now merged into this branch) and cleaned it up a bit in https://github.com/getsentry/sentry/pull/74744/commits/76d43b7b307fae6b9ffd6ad6a9d03eb04eb885f8. The level of log coverage that @iamrajjoshi implemented for Slack is now present for all providers.